storage/innobase/include/buf0buf.h

```diff
@@ -2351,8 +2351,11 @@ Use these instead of accessing buf_pool->mutex directly. */
 
 
 /** Get appropriate page_hash_lock. */
-# define buf_page_hash_lock_get(buf_pool, page_id)	\
-	hash_get_lock((buf_pool)->page_hash, (page_id).fold())
+inline rw_lock_t*
+buf_page_hash_lock_get(const buf_pool_t* buf_pool, page_id_t page_id)
```

Comment:
Declare the parameter as `const page_id_t& page_id`. We do not want any copy construction to happen.

Reply:
Would it make sense to use `static inline` here? Probably it is not strictly required in C++.

Reply:
Why? `page_id_t` is small. http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const

Reply:
You make a good point about passing a `const` reference. But I think that it applies more to non-inline functions where the reference cannot be optimized away.
And we are not yet even at C++11. With C++17 or later it might be reasonable to rely on copy elision.